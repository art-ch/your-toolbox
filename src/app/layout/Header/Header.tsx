'use client';

import React from 'react';
import { PiToolboxFill } from 'react-icons/pi';

import {
  NavigationMenu,
  NavigationMenuList
} from '@/components/ui/navigation-menu';
import { AppDrawerTrigger } from './components/AppDrawerTrigger/AppDrawerTrigger';
import Link from 'next/link';

export const Header = () => {
  return (
    <NavigationMenu className="min-w-full shadow-sm py-0.5 justify-normal">
      <div className="min-w-full px-4">
        <NavigationMenuList className="min-width-full grid grid-cols-3">
          <AppDrawerTrigger />
          <Link className="justify-self-center" href="/">
            <PiToolboxFill className="text-[40px]" />
          </Link>
        </NavigationMenuList>
      </div>
    </NavigationMenu>
  );
};
