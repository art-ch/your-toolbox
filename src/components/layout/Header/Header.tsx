'use client';

import React from 'react';
import { PiToolboxFill } from 'react-icons/pi';

import {
  NavigationMenu,
  NavigationMenuList
} from '@/components/ui/navigation-menu';
import { AppDrawerTrigger } from './components/AppDrawerTrigger/AppDrawerTrigger';
import Link from 'next/link';
import { LanguageSwitcher } from './components/LanguageSwitcher/LanguageSwitcher';
import { Language } from '@/lib/i18n/types';

export type HeaderProps = {
  language: Language;
};

export const Header = ({ language }: HeaderProps) => {
  return (
    <NavigationMenu className="min-w-full shadow-xs py-0.5 justify-normal">
      <div className="min-w-full px-4">
        <NavigationMenuList className="min-width-full grid grid-cols-3">
          <AppDrawerTrigger />
          <Link className="justify-self-center" href="/">
            <PiToolboxFill className="text-[40px]" />
          </Link>
          <LanguageSwitcher currentLanguage={language} />
        </NavigationMenuList>
      </div>
    </NavigationMenu>
  );
};
