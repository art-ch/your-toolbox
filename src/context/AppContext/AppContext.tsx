import { createContext, useContext } from 'react';
import { DrawerState, useDrawer } from './hooks/useDrawer/useDrawer';

type AppContextType = {
  drawer: DrawerState;
};

const AppContext = createContext<AppContextType | null>(null);

export const AppProvider = ({ children }: { children: React.ReactNode }) => {
  const drawer = useDrawer();

  return (
    <AppContext.Provider value={{ drawer }}>{children}</AppContext.Provider>
  );
};

export const useAppContext = () => {
  const context = useContext(AppContext);

  if (!context) {
    throw new Error('useAppContext must be used within an AppProvider');
  }

  return context;
};
