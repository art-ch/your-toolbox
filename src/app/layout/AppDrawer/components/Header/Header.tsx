'use client';

import React from 'react';
import { X } from 'lucide-react';
import Link from 'next/link';

import { useAppContext } from '@/app/context/AppContext/AppContext';

export const Header = () => {
  const { drawer } = useAppContext();

  return (
    <div className="flex justify-between min-w-full p-2.5 shadow-[1px_1px_1px_0_rgba(0,0,0,0.06)]">
      <Link href="/navigation" onClick={drawer.close}>
        <h3 className="font-semibold">Navigation</h3>
      </Link>
      <X className="cursor-pointer" onClick={drawer.close} />
    </div>
  );
};
