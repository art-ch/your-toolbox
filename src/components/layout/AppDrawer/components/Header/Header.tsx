'use client';

import React from 'react';
import { X } from 'lucide-react';
import Link from 'next/link';
import { useAppContext } from '@/context/AppContext/AppContext';
import { useTranslation } from 'react-i18next';

export const Header = () => {
  const { drawer } = useAppContext();
  const { t } = useTranslation();

  return (
    <div className="flex justify-between min-w-full p-2.5 shadow-[1px_1px_1px_0_rgba(0,0,0,0.06)]">
      <Link href="/navigation" onClick={drawer.close}>
        <h3 className="font-semibold">{t('common:navigation')}</h3>
      </Link>
      <X className="cursor-pointer" onClick={drawer.close} />
    </div>
  );
};
