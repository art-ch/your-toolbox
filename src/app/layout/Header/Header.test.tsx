import { render } from '@testing-library/react';

import { Header } from './Header';
import { AppProvider } from '@/app/context/AppContext/AppContext';

describe('Header component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <AppProvider>
        <Header />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
