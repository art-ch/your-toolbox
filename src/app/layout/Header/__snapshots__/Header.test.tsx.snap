// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <nav
    aria-label="Main"
    class="relative z-10 flex max-w-max flex-1 items-center min-w-full shadow-sm py-0.5 justify-normal"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      class="min-w-full px-4"
    >
      <div
        style="position: relative;"
      >
        <ul
          class="group flex-1 list-none items-center justify-center space-x-1 min-width-full grid grid-cols-3"
          data-orientation="horizontal"
          dir="ltr"
        >
          <svg
            class="lucide lucide-align-justify cursor-pointer"
            data-testid="app-drawer-trigger"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12h18"
            />
            <path
              d="M3 18h18"
            />
            <path
              d="M3 6h18"
            />
          </svg>
          <a
            class="justify-self-center"
            href="/"
          >
            <svg
              class="text-[40px]"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224,64H176V56a24,24,0,0,0-24-24H104A24,24,0,0,0,80,56v8H32A16,16,0,0,0,16,80v28a4,4,0,0,0,4,4H64V96.27A8.17,8.17,0,0,1,71.47,88,8,8,0,0,1,80,96v16h96V96.27A8.17,8.17,0,0,1,183.47,88,8,8,0,0,1,192,96v16h44a4,4,0,0,0,4-4V80A16,16,0,0,0,224,64Zm-64,0H96V56a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8Zm80,68v60a16,16,0,0,1-16,16H32a16,16,0,0,1-16-16V132a4,4,0,0,1,4-4H64v16a8,8,0,0,0,8.53,8A8.17,8.17,0,0,0,80,143.73V128h96v16a8,8,0,0,0,8.53,8,8.17,8.17,0,0,0,7.47-8.25V128h44A4,4,0,0,1,240,132Z"
              />
            </svg>
          </a>
        </ul>
      </div>
    </div>
    <div
      class="absolute left-0 top-full flex justify-center"
    />
  </nav>
</div>
`;
